### ISSUE TITLE
[Suggestion] Use a namedtuple for capsys results

### ISSUE THREAD

**Issue by samueldg** [pytest-dev/pytest#2879]
Currently, the `capsys` fixture's `readouterr()` method returns a tuple with (stdout, stderr) results.

Usage will look something like this:

```python
def test_print_1(capsys):
    print('something')
    out, err = capsys.readouterr()  # Named, unused variable
    assert 'something' in out

def test_print_2(capsys):
    print('something ')
    out, _ = capsys.readouterr()  # Anonymous, ignored variable
    assert 'something' in out

def test_print_3(capsys):
    print('something')
    assert 'something' in capsys.readouterr()[0]  # Index access (not super readable)
```

My suggestion would be to return a `namedtuple` instead, so the attributes can be accessed by name

```python
def test_print_4(capsys):
    print('something')
    assert 'something' in capsys.readouterr().out  # Attribute access
```

Which could be achieved by wrapping the result with something like:

```python
from collections import namedtuple
CapsysResult = namedtuple('CapsysResult', ['out', 'err'])
```

This would be pretty straightforward, and still fully compatible with existing code using tuples either for index access or unpacking.

**Comment by samueldg** [comment]
Let me know if that's something you'd want in, I'd gladly submit a PR.

**Comment by nicoddemus** [comment]
@samueldg that's a perfect case where changing to a `namedtuple` is a great idea. A PR would be greatly appreciated, thanks! (Make sure to target the `features` branch)

**Comment by samueldg** [comment]
@nicoddemus There you go: https://github.com/pytest-dev/pytest/pull/2880

**Comment by nicoddemus** [comment]
Awesome, thanks!

**Comment by RonnyPfannschmidt** [comment]
closed by #2880
